src/Cli/dotnet/CommonArguments.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System.CommandLine;
+using System.CommandLine.Parsing;
+
+namespace Microsoft.DotNet.Cli
+{
+    internal class CommonArguments
+    {
+        #region PackageIdentityArgument
+        public static CliArgument<(string PackageId, string Version)> PackageIdentityArgument(bool requireArgument = true) =>
```

Comment:
It is a general code smell to publicly expose ValueTuples. The use of it in the private method below is fine, but I wouldn't let them escape the class scope. I'd just make a quick record that represents this.

The rationale is that every use of the ValueTuple needs to be updated if you need to add values or do anything with it. It is fine as a transient bundle of values, but once it hits consumption outside of the class, it should be a proper type if it is easy to do so.

Reply:
strong agree - we should just use the NuGet PackageIdentity type here. it already captures the idea of a package identifier + optional version.

Reply:
Ohh that makes a lot of sense, will look into that, thx!